### PR TITLE
Link to mypy's error code documentation

### LINKED ISSUE #57: Feature request: Link to the error's documentation
I would like for the error info popup to also link to the appropriate error code documentation. For example `no-untyped-def` would link to https://mypy.readthedocs.io/en/stable/error_code_list2.html#check-that-every-function-has-an-annotation-no-untyped-def . Similarly to how ESLint and Pylance do it: 
image
image

I found that it is an extremely powerful tool for learning and help devs understand the errors. Especially newer developers who are less keen to go dig up documentation they don't even know if it exists.

### PR DESCRIPTION
**Pull request by Avasam**
Closes #57

~~I wish there was a better way to map code to url for maintainability, I have opened a doc feature request with mypy to have better way to programmatically link to the codes' doc. Both for better anchors and a way that doesn't span 2 pages.~~ https://github.com/python/mypy/issues/15431

Demo:
image
Which links to https://mypy.readthedocs.io/en/latest/_refs.html#code-import

Other potential improvements:
- We could link to a specific mypy version doc instead of "stable/latest". But I don't it would bring much value anyway.
- After the next mypy release, "latest" can be changed to "stable".

### CODE REVIEW

File: bundled/tool/lsp_server.py [hunk 1]
```diff
@@ -142,6 +142,14 @@ def _get_group_dict(line: str) -> Optional[Dict[str, str]]:
     return None
 
 
+def _build_message_doc_url(code: str) -> str:
+    """Build the URL to the documentation for this warning."""
+    anchor = utils.ERROR_CODE_ANCHORS.get(code)
+    if anchor is None:
+        return ""
+    return utils.ERROR_CODE_BASE_URL + anchor
```
Comment (karthiknadig) [inline bundled/tool/lsp_server.py]:
```suggestion
def _get_code_description(code: str) -> Opinion[lsp.CodeDescription]:
    """Build the URL to the documentation for this warning."""
    anchor = utils.ERROR_CODE_ANCHORS.get(code)
    if anchor is None:
        return None
    return lsp.CodeDescription(href=utils.ERROR_CODE_BASE_URL + anchor)
```

File: bundled/tool/lsp_server.py [hunk 2]
```diff
@@ -174,6 +182,8 @@ def _parse_output_using_regex(
                     character=end_column,  # ignore col_offset for end
                 )
 
+            documentation_url = _build_message_doc_url(data["code"])
+
```
Comment (karthiknadig) [inline bundled/tool/lsp_server.py]:
```suggestion
```

File: bundled/tool/lsp_server.py [hunk 3]
```diff
@@ -182,6 +192,7 @@ def _parse_output_using_regex(
                 message=data.get("message"),
                 severity=_get_severity(data["code"], data["type"], severity),
                 code=data["code"],
+                code_description=lsp.CodeDescription(href=documentation_url),
```
Comment (karthiknadig) [inline bundled/tool/lsp_server.py]:
```suggestion
                code_description=_get_code_description(data["code"]),
```

**Review by karthiknadig — CHANGES_REQUESTED** [review]
Thanks for the PR 🥳

I have some minor changes. I think we can directly return a `CodeDescription` from the helper function. That would also mean that the case where we don't have a link will give the same old result as before.

File: bundled/tool/lsp_server.py (hunk 1 above)
Comment (Avasam) [reply bundled/tool/lsp_server.py]: I think you meant `Optional` ?

```suggestion
def _get_code_description(code: str) -> Optional[lsp.CodeDescription]:
    """Build the URL to the documentation for this warning."""
    anchor = utils.ERROR_CODE_ANCHORS.get(code)
    if anchor is None:
        return None
    return lsp.CodeDescription(href=utils.ERROR_CODE_BASE_URL + anchor)
```

**Review by karthiknadig — DISMISSED** [review]

**Comment by karthiknadig** [comment]
@Avasam Looks like there is a formatting issue. Can you run `black` on `lsp_utils.py`?

**Comment by Avasam** [comment]
With https://github.com/python/mypy/pull/15435 and https://github.com/python/mypy/pull/15437, up to you if you wanna update the logic in a separate PR or wait for the next mypy release so we can point to stable in this PR using consistent anchors.

**Comment by karthiknadig** [comment]
@Avasam We can point to the latest. With the assumption that the error messages are not going to change significantly.

**Review by anthonykim1 — DISMISSED** [review]

**Review by karthiknadig — APPROVED** [review]

**Review by anthonykim1 — APPROVED** [review]

**Pull request merged**